### PR TITLE
fix: 488 Fix weighing time redundancy analyses

### PR DESCRIPTION
**Pull request by ArdtK**
## Issue addressed
Solves #488, introduced by #433.

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [ ] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
Instead of calculating the distance first and based on that calculate the alternative value (distance or time), it now directly calculates the alternative value (and path) based on the weighing.

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [ ] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
The dijkstra calls for getting the optimal route (length and nodes) are combined in 1 call.

### CODE REVIEW

File: ra2ce/analysis/losses/weighing_analysis/weighing_analysis_protocol.py
Comment (ArdtK) [inline ra2ce/analysis/losses/weighing_analysis/weighing_analysis_protocol.py:1]: @Carsopre I didn't add the dijkstra calculation to the weighing analyser, as it calls the same method for both "length" and "time". With the removal of this method, I doubt if having these weighing analysers still are valuable.
I'm curious to hear your opinion.

File: ra2ce/analysis/losses/multi_link_redundancy.py [hunk 1]
```diff
@@ -152,27 +152,28 @@ def _is_not_none(value):
...
-                    alt_value = _weighing_analyser.calculate_alternative_value(alt_dist)
+                    _connected = 1
 
-                    diff = round(alt_value - _current_value, 3)
+                    diff = round(_alt_value - _current_value, 3)
                 else:
-                    alt_value = _current_value
-                    alt_nodes, connected = np.NaN, 0
+                    _alt_value = _current_value
+                    _alt_nodes, _connected = np.NaN, 0
```
Comment (sahand-asgarpour) [inline ra2ce/analysis/losses/multi_link_redundancy.py:166]: `_alt_value, _alt_nodes, _connected = np.NaN, np.NaN, 0 `

File: ra2ce/analysis/losses/single_link_redundancy.py [hunk 2]
```diff
@@ -68,15 +68,13 @@ def execute(self) -> GeoDataFrame:
...
                     self.graph_file.graph,
                     u,
                     v,
                     weight=self.analysis.weighing.config_value,
                 )
-                _alt_nodes = nx.dijkstra_path(self.graph_file.graph, u, v)
-                _alt_value = _weighing_analyser.calculate_alternative_value(_alt_dist)
 
                 # append alternative route nodes
                 _alt_value_list.append(_alt_value)
```
Comment (sahand-asgarpour) [inline ra2ce/analysis/losses/single_link_redundancy.py:80]: @ArdtK 
_alt_value_list.append(math.nan)

File: ra2ce/analysis/losses/multi_link_redundancy.py [hunk 3]
```diff
@@ -152,27 +152,28 @@ def _is_not_none(value):
...
 
-                    diff = round(alt_value - _current_value, 3)
+                    diff = round(_alt_value - _current_value, 3)
                 else:
-                    alt_value = _current_value
-                    alt_nodes, connected = np.NaN, 0
-                    diff = np.NaN
+                    _alt_value = np.nan
+                    _alt_nodes, _connected = np.nan, 0
+                    diff = np.nan
```
Comment (Carsopre) [inline ra2ce/analysis/losses/multi_link_redundancy.py:167]: it makes more sense defining the values beforehand (fallback / default values) then overwriting them in the if.

**Review by Carsopre — APPROVED** [review]
I'm assuming it's okay, yet I do not understand Sahand's comment on appending a `nan` value rather than the current `_alt_value` (`single_link_redundancy.py::80`)

File: ra2ce/analysis/losses/multi_link_redundancy.py (hunk 3 above)
Comment (ArdtK) [reply ra2ce/analysis/losses/multi_link_redundancy.py:167]: Done in both analyses

**Pull request merged**